src/consumer/kafka.py

```
@@ -0,0 +1,325 @@
+                try:
+                    # Poll with timeout (milliseconds)
+                    message_batch = await self.consumer.getmany(timeout_ms=1000)
+
+                    if not message_batch:
+                        continue
+
+                    # Process messages from all partitions
+                    for tp, messages in message_batch.items():
+                        logger.info(f"Received {len(messages)} messages from {tp.topic}:{tp.partition}")
+
+                        topic = msg.topic
```

Ruff (0.8.2)

Undefined name `msg`

(F821)